### PR TITLE
Implement new merchant product api endpoint

### PR DESCRIPTION
**Pull request by PaddyS**

### CODE REVIEW

File: src/Merchants/Content/Merchant/Api/MerchantProductController.php [hunk 1]
```diff
@@ -0,0 +1,314 @@
+<?php declare(strict_types=1);
```
Comment (PaddyS) [inline src/Merchants/Content/Merchant/Api/MerchantProductController.php]: @shyim Should I move this class somewhere else? Not too sure about the destination

Comment (shyim) [reply src/Merchants/Content/Merchant/Api/MerchantProductController.php]: Should be okay :)

File: src/Merchants/Content/Merchant/Api/MerchantProductController.php [hunk 2]
```diff
@@ -0,0 +1,315 @@
...
+                'tax' => $product->getTax()->getTaxRate(),
+            ];
+
+            if ($product->getCover()) {
+                $productData['media'] = $product->getCover()->getMedia()->getUrl();
+            }
+            $products[] = $productData;
+        }
+
+        dd($products);
```
Comment (shyim) [inline src/Merchants/Content/Merchant/Api/MerchantProductController.php]: dd

File: src/Merchants/Content/Merchant/Api/MerchantProductController.php [hunk 3]
```diff
@@ -0,0 +1,315 @@
...
+        $criteria = new Criteria([$merchantId]);
+        $criteria->addAssociation('products');
+        $criteria->addAssociation('products.cover');
+
+        $merchant = $this->merchantRepository->search($criteria, $context->getContext())->first();
+
+        $products = [];
+        /** @var ProductEntity $product */
+        foreach ($merchant->getProducts() as $key => $product) {
+            $productData = [
```
Comment (shyim) [inline src/Merchants/Content/Merchant/Api/MerchantProductController.php]: why not just encode the product 😂

Comment (PaddyS) [reply src/Merchants/Content/Merchant/Api/MerchantProductController.php]: Do we want to expose **everything**?

Comment (shyim) [reply src/Merchants/Content/Merchant/Api/MerchantProductController.php]: Good argument

**Pull request merged**